### PR TITLE
block factory

### PR DESCRIPTION
**Pull request by kecajjo**
I'll talk to you on Monday about other DE and IWYU, meanwhile - block factory PR

### CODE REVIEW

File: bl/blocks/block_factory/src/block_factory.cpp [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+            break;
+        case BlockType::LSHAPE:
+            return std::make_unique<LShape>();
+            break;
+        case BlockType::TSHAPE:
+            return std::make_unique<TShape>();
+            break;
+        case BlockType::SQUARE:
+            return std::make_unique<Square>();
+            break;
```
Comment (ProrokWielki) [inline bl/blocks/block_factory/src/block_factory.cpp]: you dont need breaks after return

File: bl/blocks/block_factory/inc/block_factory.hpp [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+#include <memory>
+
+#include "block.hpp"
+
+enum class BlockType { DIAGONAL, LSHAPE, TSHAPE, SQUARE };
+
+class BlockFactory {
+   public:
+    BlockFactory() = delete;
+    static std::unique_ptr<Block> createBlock(BlockType type);
```
Comment (ProrokWielki) [inline bl/blocks/block_factory/inc/block_factory.hpp:12]: I'm pretty sure you'll change it to shared pointer later on, but if you want to keep it I'm ok with that :P

**Review by ProrokWielki — CHANGES_REQUESTED** [review]

File: bl/blocks/block_factory/inc/block_factory.hpp (hunk 2 above)
Comment (kecajjo) [reply bl/blocks/block_factory/inc/block_factory.hpp:12]: I'll change it shared_ptr later on, but conversion from unique_ptr is possible while other way around it's not, so it's more versatile this way :D

File: bl/blocks/block_factory/src/block_factory.cpp (hunk 1 above)
Comment (kecajjo) [reply bl/blocks/block_factory/src/block_factory.cpp]: fixed

**Review by ProrokWielki — APPROVED** [review]

**Pull request merged**